File: src/AnQL.Core/AnQL.cs
using Antlr4.Runtime;

namespace AnQL.Core;

public static class AnQL
{
    public static AnQLGrammarParser BuildParser(string query)
    {
        return new AnQLGrammarParser(new CommonTokenStream(new AnQLGrammarLexer(new AntlrInputStream(query.ToLower()))));
    }
}

File: typescript/core/src/AnQL.ts
import { CharStreams, CommonTokenStream } from 'antlr4ts'

import { AnQLGrammarParser } from '@gen/grammar/AnQLGrammarParser'
import { AnQLGrammarLexer } from '@gen/grammar/AnQLGrammarLexer'

export function buildGrammarParser(query: string): AnQLGrammarParser {
    return new AnQLGrammarParser(new CommonTokenStream(new AnQLGrammarLexer(CharStreams.fromString(query))));
}